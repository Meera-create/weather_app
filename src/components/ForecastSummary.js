import React from "react";
import iconData from '../data/iconData.json'



function ForecastSummary(props){

    const{
        date,
        description,
        icon,
        temperature
    }=props;
    //console.log(props,"heeeeloooJGHGJHGJH")




    const weatherCode = icon.slice(0,1) + "00";
    const formattedDate=new Date(date).toDateString();

    return (
    <div className='forecast-summary' data-testid='forecast-summary'>
        <div className='forecast-summary_date'>
        {formattedDate}
        </div>
        
        <div className='forecast-summary_description'>
        {description}
        </div>
        
        <div className='forecast-summary_icon' data-testid='forecast-icon'>
            <img src={iconData[weatherCode]}/>
            </div>
        
        <div className='forecast-summary_temperature'>
            {temperature.max}
            &deg;C
        </div>
        <button type="button" onClick={()=> onselect(date)}>More Details</button>
    </div>
    );
};


export default ForecastSummary