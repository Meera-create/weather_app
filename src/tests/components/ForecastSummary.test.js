import React from "react";
import ForecastSummary from "../../components/ForecastSummary";
import {getByTestId, render} from "@testing-library/react"

describe('ForecastSummary',() => {
    const validProps = {
      
        date:111111,
        description:'Stub description',
        icon:'StubIcon',
        temperature: {
            min: 12,
            max: 22,
        },
    };

    it("renders correctly",() => {
        const {asFragment} = render(
            <ForecastSummary
                date={validProps.date}
                description={validProps.description}
                icon={validProps.icon}
                temperature={validProps.temperature}
            />
        );
    expect(asFragment()).toMatchSnapshot();
})

    it("renders correct values for props",()=>{
        const {getByText,getByTestId} = render(
            <ForecastSummary
                date={validProps.date}
                description={validProps.description}
                icon={validProps.icon}
                temperature={validProps.temperature}
                
            />
        );
    expect(getByText("Thu Jan 01 1970")).toHaveAttribute("class", "forecast-summary_date");
    expect(getByText("Stub description")).toHaveAttribute("class","forecast-summary_description");
    expect(getByTestId("forecast-icon")).toHaveAttribute("class","forecast-summary_icon");
    expect(getByText("22Â°C")).toHaveAttribute("class","forecast-summary_temperature")

    })

   
});