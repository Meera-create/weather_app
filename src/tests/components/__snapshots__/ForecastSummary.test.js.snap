// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`ForecastSummary renders correctly 1`] = `
<DocumentFragment>
  <div
    class="forecast-summary"
    data-testid="forecast-summary"
  >
    <div
      class="forecast-summary_date"
    >
      Thu Jan 01 1970
    </div>
    <div
      class="forecast-summary_description"
    >
      Stub description
    </div>
    <div
      class="forecast-summary_icon"
      data-testid="forecast-icon"
    >
      <img />
    </div>
    <div
      class="forecast-summary_temperature"
    >
      22Â°C
    </div>
    <button
      type="button"
    >
      More Details
    </button>
  </div>
</DocumentFragment>
`;
